akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1338,3 +1338,36 @@ abstract class AbstractOutHandler extends OutHandler
  * (completing when upstream completes, failing when upstream fails, completing when downstream cancels).
  */
 abstract class AbstractInOutHandler extends InHandler with OutHandler
+
+private[akka] trait CallbackWrapper[T] {
```

Comment:
Is this really general enough to not live together with the `QueueSource` but with the generic handler code? If so it should not contain an error naming `SourceQueue.offer` below.